### PR TITLE
Clarify dependencies for context

### PR DESCRIPTION
**Pull request by live1206**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6060)

### CODE REVIEW

File: src/docfx/lib/schema/JsonSchemaTransformer.cs [hunk 1]
```diff
@@ -33,47 +33,72 @@ public JsonSchemaTransformer(JsonSchema schema)
...
             var uidCount = _uidCountCache.GetOrAdd(file, GetFileUidCount(_schema, token));
-            return TransformContentCore(file, context, _schema, token, uidCount);
+            return TransformContentCore(file, _schema, token, uidCount, context.MarkdownEngine, context.LinkResolver, context.XrefResolver);
         }
 
-        public (List<Error>, IReadOnlyList<InternalXrefSpec>) LoadXrefSpecs(Document file, Context context, JToken token)
+        public (List<Error>, IReadOnlyList<InternalXrefSpec>) LoadXrefSpecs(
+            Document file,
+            JToken token,
+            MarkdownEngine markdownEngine,
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchemaTransformer.cs:42]: Move these services to the constructor of `JsonSchemaTransformer`?

Comment (live1206) [reply src/docfx/lib/schema/JsonSchemaTransformer.cs:42]: I could give it a try, ideally we should not pass around the component instances, but `JsonSchemaTransformer` is not hold in `context`.

File: src/docfx/build/context/Context.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Net.Mail;
```
Comment (yufeih) [inline src/docfx/build/context/Context.cs:5]:
```suggestion
~~using System.Net.Mail;~~
```

**Pull request merged**